stripe/src/main/java/com/stripe/android/IssuingCardPinService.java

```diff
@@ -0,0 +1,268 @@
+     * @param userOneTimeCode the one-time code that was sent to the cardholder through sms or email
+     * @param listener        a listener for either the PIN, or any error that can occur
+     */
+    public void retrievePin(
+            @NonNull String cardId,
+            @NonNull String verificationId,
+            @NonNull String userOneTimeCode,
+            @Nullable IssuingCardPinRetrievalListener listener
+    ) {
+        mCardPinUpdateListener = null;
+        mCardPinRetrievalListener = listener;
+
+        Map<String, Object> arguments = new HashMap<>();
+        arguments.put("cardId", cardId);
+        arguments.put("verificationId", verificationId);
+        arguments.put("userOneTimeCode", userOneTimeCode);
+
+        mEphemeralKeyManager.retrieveEphemeralKey(PIN_RETRIEVE, arguments);
+    }
+
+    /**
+     * Retrieves a PIN for a given card
+     *
+     * @param cardId          the ID of the card (looks like ic_abcdef1234)
+     * @param newPin          the new desired PIN
+     * @param verificationId  the ID of the verification that was sent to the cardholder
+     *                        (typically server-side, through /v1/issuing/verifications)
+     * @param userOneTimeCode the one-time code that was sent to the cardholder through sms or email
+     * @param listener        a listener for either the PIN, or any error that can occur
+     */
+    public void updatePin(
+            @NonNull String cardId,
+            @NonNull String newPin,
+            @NonNull String verificationId,
+            @NonNull String userOneTimeCode,
+            @Nullable IssuingCardPinUpdateListener listener
+    ) {
+        mCardPinRetrievalListener = null;
+        mCardPinUpdateListener = listener;
+
+        Map<String, Object> arguments = new HashMap<>();
+        arguments.put("cardId", cardId);
+        arguments.put("newPin", newPin);
+        arguments.put("verificationId", verificationId);
+        arguments.put("userOneTimeCode", userOneTimeCode);
+
+        mEphemeralKeyManager.retrieveEphemeralKey(PIN_UPDATE, arguments);
+    }
+
+    @Override
+    public void onKeyUpdate(
+            @Nullable IssuingCardEphemeralKey ephemeralKey,
+            @Nullable String action,
+            @Nullable Map<String, Object> arguments
+    ) {
+
+        if (PIN_RETRIEVE.equals(action)) {
+            String cardId = (String) arguments.get("cardId");
+            String verificationId = (String) arguments.get("verificationId");
+            String userOneTimeCode = (String) arguments.get("userOneTimeCode");
+            try {
+                String pin = StripeApiHandler.retrieveIssuingCardPin(
+                        cardId,
+                        verificationId,
+                        userOneTimeCode,
+                        ephemeralKey.getSecret());
+
+                if (mCardPinRetrievalListener != null) {
+                    mCardPinRetrievalListener.onIssuingCardPinRetrieved(pin);
```

Comment:
`pin` is currently `@Nullable`, while `onIssuingCardPinRetrieved()` expects a `@NonNull`

Reply:
👍